test_quality_of_service/test/test_deadline.cpp

```diff
@@ -0,0 +1,167 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <memory>
+#include <string>
+#include <tuple>
+
+#include "gtest/gtest.h"
+
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp/executors/single_threaded_executor.hpp"
+
+#include "std_msgs/msg/string.hpp"
+
+#include "test_quality_of_service/qos_test_publisher.hpp"
+#include "test_quality_of_service/qos_test_subscriber.hpp"
+#include "test_quality_of_service/qos_utilities.hpp"
+
+using namespace std::chrono_literals;
+
+/// Test Deadline with a single subscriber node
+TEST_F(QosRclcppTestFixture, test_deadline_no_publisher) {
+  const std::chrono::milliseconds deadline_duration = 1s;
+  const std::chrono::milliseconds test_duration = 10500ms;
+  const std::tuple<size_t, size_t> deadline_duration_tuple = convert_chrono_milliseconds_to_size_t(
+    deadline_duration);
+  const int expected_number_of_deadline_callbacks = test_duration / deadline_duration;
+
+  int total_number_of_subscriber_deadline_events = 0;
+  int last_sub_count = 0;
+
+  rmw_qos_profile_t qos_profile = rmw_qos_profile_default;
+  qos_profile.deadline.sec = std::get<0>(deadline_duration_tuple);
```

Comment:
`std::tie(qos_profile.deadline.sec, qos_profile.deadline.nsec) = deadline_duration_tuple`
may work

Reply:
Yup! Implemented in latest changes.